src/names.c

```diff
@@ -90,102 +95,95 @@ SEXP as_unique_names(SEXP names, bool quiet) {
   }
 
   R_len_t i = 0;
-  R_len_t n = Rf_length(names);
-  SEXP* ptr = STRING_PTR(names);
+  const R_len_t n = Rf_length(names);
```

Comment:
Is there a practical reason to use `const` here, such as better assembly output? I'd rather not overdo const-qualifiers as a matter of style.

Reply:
It does ease the burden on the optimizer, but this is not the point.

It's the guarantee that the value of `n` doesn't change, in line with pure functional programming style. Anyone reading the code immediately knows what the value of `n` must be at any occurrence. Occasionally this style will catch errors, when a typo would lead to updating the wrong variable.

It's too bad that the `const` adds so much noise. Maybe with typedefs, or written as `R_len_t const n = ...` ?

Reply:
I don't think this is important and that we should use `const` everywhere just because we can. This overall hinders reading code. I'd rather use it with data arrays, and where it actually helps the compiler.